core/src/main/java/io/questdb/cairo/TableReader.java

```diff
@@ -206,13 +206,29 @@ public BitmapIndexReader getBitmapIndexReader(int partitionIndex, int columnInde
 
     public BitmapIndexReader getBitmapIndexReader(int partitionIndex, int columnBase, int columnIndex, int direction) {
         final int index = getPrimaryColumnIndex(columnBase, columnIndex);
+        final long partitionTimestamp = txFile.getPartitionTimestamp(partitionIndex);
+        final long columnNameTxn = columnVersionReader.getColumnNameTxn(partitionTimestamp, metadata.getWriterIndex(columnIndex));
+        final long partitionTxn = txFile.getPartitionNameTxn(partitionIndex);
+
         BitmapIndexReader reader = bitmapIndexes.getQuick(direction == BitmapIndexReader.DIR_BACKWARD ? index : index + 1);
         if (reader != null) {
+            // make sure to reload the reader
+            final String columnName = metadata.getColumnName(columnIndex);
+            final long columnTop = getColumnTop(columnBase, columnIndex);
+            try {
+                Path path = pathGenPartitioned(partitionIndex);
+                if (reader instanceof BitmapIndexBwdReader) {
+                    ((BitmapIndexBwdReader) reader).of(configuration, path, columnName, columnNameTxn, columnTop, partitionTxn);
```

Comment:
can we extract this method on the interface to avoid `instanceOf` ?

Reply:
Good idea. Done in ad5a21c6b8edf3366ae3b2aae826683dcd43ea0f